pkg/api/types.go

```diff
@@ -665,8 +665,10 @@ type EnvVar struct {
 
 // EnvVarSource represents a source for the value of an EnvVar.
 type EnvVarSource struct {
-	// Required: Selects a field of the pod; only name and namespace are supported.
-	FieldRef *ObjectFieldSelector `json:"fieldRef"`
+	// Selects a field of the pod; only name and namespace are supported.
+	FieldRef *ObjectFieldSelector `json:"fieldRef,omitempty"`
```

Comment:
In theory, yes. Clearly, it was intended to be optional. 
a) It's a pointer.
b) There would be no point in the extra layer of struct if we didn't intend to add other fields.

v1/types.go doesn't say "Required", but also lacks omitempty. :-(

I'd be in favor of fixing the bug, but ask on kubernetes-dev whether it would affect anyone and add a release note.